src/ctx_macro.rs

```diff
@@ -0,0 +1,239 @@
+/// let mut buf1 = [0u8; 32];
+/// let mut buf2 = [0u8; 64];
+/// for offset in [0, 8, 16, 24] {
+///     case_set!(16, buf=[&mut buf1[..], &mut buf2[..]], len=[8, 16], offset=[offset, offset*2], {
+///         set!(buf, len as u8 >> 3);
+///     });
+/// }
+/// ```
+/// 
+/// A more realistic example of filling multiple buffers with the same value is
+/// initializing different struct fields at the same time (from
+/// `src/decode.rs`):
+/// ```ignore
+/// case_set!(32, ctx=[(&t.l, 1), (&f.a[t.a], 0)], len=[bh4, bw4], offset=[by4, bx4], {
+///      let (dir, dir_index) = ctx;
+///      set_disjoint!(dir.seg_pred, seg_pred.into());
+///      set_disjoint!(dir.skip_mode, b.skip_mode);
+///      set_disjoint!(dir.intra, 0);
+///      set_disjoint!(dir.skip, b.skip);
+///      set_disjoint!(dir.pal_sz, 0);
+/// });
+/// ```
+/// 
+/// [`DisjointMut`]: crate::src::disjoint_mut::DisjointMut
+#[macro_export]
+macro_rules! case_set {
+    ($UP_TO:literal, $($WITH_DEFAULT:literal,)? $ctx:ident=[$($ctx_expr:expr),*], $len:ident=[$($len_expr:expr),*], $offset:ident=[$($offset_expr:expr),*], $body:block) => {
+        for ($ctx, ($len, $offset)) in core::iter::zip([$($ctx_expr,)*], core::iter::zip([$($len_expr,)*], [$($offset_expr,)*])) {
+            case_set!($UP_TO, $($WITH_DEFAULT,)? $ctx=$ctx, $len=$len, $offset=$offset, $body);
+        }
+    };
+    ($UP_TO:literal, $($WITH_DEFAULT:literal,)? $len:ident, $offset:ident, $body:block) => {
+        case_set!($UP_TO, $($WITH_DEFAULT,)? ctx=(), $len=$len, $offset=$offset, $body);
+    };
+    ($UP_TO:literal, $($WITH_DEFAULT:literal,)? $len:ident=$len_expr:expr, $offset:ident, $body:block) => {
+        case_set!($UP_TO, $($WITH_DEFAULT,)? ctx=(), $len=$len_expr, $offset=$offset, { $(($buf, $val)),* });
+    };
+    ($UP_TO:literal, $($WITH_DEFAULT:literal,)? $len:ident, $offset:ident=$offset_expr:expr, $body:block) => {
+        case_set!($UP_TO, $($WITH_DEFAULT,)? ctx=(), $len=$len, $offset=$offset_expr, $body);
+    };
+    ($UP_TO:literal, $($WITH_DEFAULT:literal,)? $len:ident=$len_expr:expr, $offset:ident=$offset_expr:expr, $body:block) => {
+        case_set!($UP_TO, $($WITH_DEFAULT,)? ctx=(), $len=$len_expr, $offset=$offset_expr, $body);
+    };
+    ($UP_TO:literal, $($WITH_DEFAULT:literal,)? $ctx:ident=$ctx_expr:expr, $len:ident=$len_expr:expr, $offset:ident=$offset_expr:expr, $body:block) => {
+        let $ctx = $ctx_expr;
+        let $len = $len_expr;
+        let $offset = $offset_expr;
+        {
+            #[allow(unused_macros)]
+            macro_rules! set {
+                ($buf:expr, $val:expr) => {
+                    assert!($offset <= $buf.len() && $offset + $len <= $buf.len());
+                };
+            }
+            #[allow(unused_macros)]
+            macro_rules! set_disjoint {
+                ($buf:expr, $val:expr) => {
+                    assert!($offset <= $buf.len() && $offset + $len <= $buf.len());
+                };
+            }
```

Comment:
```suggestion
            use set as set_disjoint;
```

Since they're identical.